dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java

```diff
@@ -0,0 +1,164 @@
+        removeAnyExistingJob();
+
+        //Now enqueue a few jos .. They will sleep randomly to simulate work.
+        for (int i = 1; i <= MyStatefulJob.MAX_THREADS; i++) {
+            MyStatefulJob.fireJob(ImmutableMap.of("index", i));
+            //Now verify the detail has been added for the present trigger we're introducing
+            final Optional<Map<String, Object>> triggerJobDetail = DotStatefulJob
+                    .getTriggerJobDetail(MyStatefulJob.class);
+            assertNotNull(triggerJobDetail);
+            assertTrue(triggerJobDetail.isPresent());
+            final Map<String, Object> detail = triggerJobDetail.get();
+            assertNotNull(detail);
+            //The size of the detail should have grown as we insert new triggers.
+            assertEquals(detail.size(), i);
+        }
+
+        //Verify at least 1 job has been launched.
+        final Optional<JobExecutionContext> jobExecutionContext = getJobExecutionContext();
+        assertTrue(jobExecutionContext.isPresent());
+
+        //Now lets wait for the all the threads to complete.
+        MyStatefulJob.countDownLatch.await();
+
+        //Now lets revise the execution times. Since they were supposed to run sequentially the should never overlap.
+        final HashSet<MyStatefulJob> myStatefulJobs = new HashSet<>(MyStatefulJob.finishedJobs);
+        final Iterator<MyStatefulJob> iterator = myStatefulJobs.iterator();
+        while(iterator.hasNext()){
+            final MyStatefulJob myStatefulJob = iterator.next();
+            iterator.forEachRemaining(job -> {
+                //They should never overlap.
+                assertFalse(myStatefulJob.getTimeRange().overlaps(job.getTimeRange()));
+            });
+        }
+    }
+
+    /**
+     * This gets you the execution context for the given job-name.
+     * @return
+     */
+    private Optional<JobExecutionContext> getJobExecutionContext(){
+        final String jobName = DotStatefulJob.getJobName(MyStatefulJob.class);
+        try {
+            final Scheduler sequentialScheduler = getSequentialScheduler();
+            @SuppressWarnings("unchecked")
+            final List<JobExecutionContext> executingJobs = sequentialScheduler.getCurrentlyExecutingJobs();
+            return executingJobs.stream().filter(jobExecutionContext -> {
+                final JobDetail jobDetail = jobExecutionContext.getJobDetail();
+                return jobDetail != null && jobName.equals(jobDetail.getName());
+            }).findFirst();
+        } catch (Exception e) {
+            Logger.error(DotStatefulJobTest.class, "Error retrieving execution context. " , e);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * additional struct to follow execution time lapse.
+     */
+    public static class LocalTimeRange {
+
+        private final LocalTime from;
+        private final LocalTime to;
+
+        LocalTimeRange(final LocalTime from, final LocalTime to) {
```

Comment:
Codacy found an issue: [Avoid variables with short names like to](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6077927)